frontend/src/component/Activity/Preview.jsx

```diff
@@ -0,0 +1,222 @@
+      method: 'GET'
+    })
+
+    if (!fetchResult.ok) {
+      this.handleUnavailablePreview()
+      return
+    }
+
+    const htmlCode = await fetchResult.text()
+
+    return (
+      <div className='activityFeed__preview__html'>
+        <article
+          dangerouslySetInnerHTML={{
+            __html: removeInteractiveContentFromHTML(htmlCode)
+          }}
+        />
+      </div>
+    )
+  }
+
+  getJPEGPreview (previewUrl) {
+    const { content } = this.props
+    const label = content.label
+    const filenameNoExtension = removeExtensionOfFilename(content.filename)
+
+    const prev = (width) => (
+      jpgPreviewUrl(
+        FETCH_CONFIG.apiUrl,
+        content.workspace_id,
+        content.content_id,
+        content.current_revision_id,
+        filenameNoExtension,
+        1,
+        width,
+        MAX_PREVIEW_HEIGHT
+      )
+    )
+
+    const src = ([mediaQuery, width]) => prev(width) + ' ' + width + 'w'
+
+    return (
+      <div class='activityFeed__preview__image'>
+        <img
+          alt={this.props.t('Preview of {{content}}', { content: label })}
+          title={label}
+          src={prev(PREVIEW_WIDTHS[0][1])} // fall back on the smallest image size
+          srcset={PREVIEW_WIDTHS.map(src).join(',')}
+          sizes={
+            PREVIEW_WIDTHS
+              .map(([mediaQuery, width]) => `${mediaQuery} ${width}px`.trim())
+              .join(',')
+          }
+          onError={() => this.handleUnavailablePreview()}
+        />
+      </div>
+    )
+  }
+
+  setPreview (preview) {
+    this.setState({
+      preview,
+      previewLoading: false
+    })
+  }
+
+  async updatePreview () {
+    let preview = null
+
+    if (this.isHtmlPreview()) {
+      if (!this.state.preview) {
+        this.setLoadingPreview()
+      }
+
+      try {
+        preview = await this.getHTMLPreview()
+      } catch (e) {
+        this.handleUnavailablePreview()
+        console.error('Unable to produce the preview of content ', this.props.content, e)
+        return
+      }
+
+    } else {
+      preview = this.getJPEGPreview()
+    }
+
+    this.setPreview(preview)
+  }
+
+  receivePreviewRef (ref) {
+    this.previewRef = ref
+    this.testPreviewOverflow()
+  }
+
+  componentDidUpdate (prevProps) {
+    if (prevProps.content === this.props.content) {
+      this.testPreviewOverflow()
+    } else if (prevProps.content.current_revision_id !== this.props.content.current_revision_id) {
+      this.updatePreview()
+    }
+  }
+
+  componentDidMount () {
+    if (this.isHtmlPreview()) {
+      this.updatePreview()
+    } else {
+      this.testPreviewOverflow()
+    }
+  }
+
+  testPreviewOverflow () {
+    const ref = this.previewRef
+    if (ref) {
+      const previewOverflow = ref.scrollHeight > ref.offsetHeight
+      if (this.state.previewOverflow !== previewOverflow) {
+        this.setState({ previewOverflow })
+      }
+    }
+  }
+
+  render () {
+    const { props } = this
```

Comment:
```suggestion
    const { props, state } = this
```
and update the rest of the function...